### PR TITLE
Add cbrt function for intervals

### LINKED ISSUE #310: Base.cbrt() causes stack overflow on intervals
how to reproduce
```
julia> cbrt(Interval(1,2))
ERROR: StackOverflowError:
Stacktrace:
 [1] cbrt(::Interval{Float64}) at ./special/cbrt.jl:34 (repeats 80000 times)
```
This happens due to an unfortunate combination of facts:
* `float(Interval(1,2))` returns an `Interval`
* Intervals have supertype `Real`
* Base expects that `float()` returns something a subtype of `AbstractFloat`

possible fixes:
* change supertype of Interval to `AbstractFloat`
* specialize `cbrt()` for Intervals

### PR DESCRIPTION
**Pull request by krish8484**
Fixes #310

### CODE REVIEW

File: src/intervals/functions.jl [hunk 1]
```diff
@@ -233,6 +233,15 @@ function sqrt(a::Interval{T}) where T
     @round(sqrt(a.lo), sqrt(a.hi))  # `sqrt` is correctly-rounded
 end
 
+
+
+function cbrt(a::Interval{T}) where T
+    isempty(a) && return a
+
+    Interval(cbrt(a.lo), cbrt(a.hi))
```
Comment (dpsanders) [inline src/intervals/functions.jl:241]: This does not take account of directed rounding.

Comment (dpsanders) [reply src/intervals/functions.jl:241]: e.g. try `cbrt(1..3)` instead and compare to the result when you use a BigFloat interval.

Comment (krish8484) [reply src/intervals/functions.jl:241]: We will need a similar function like sqrt_round in [FastRounding.jl](https://github.com/JeffreySarnoff/FastRounding.jl/blob/master/src/FastRounding.jl), right?

Comment (lbenet) [reply src/intervals/functions.jl:241]: I think you can mimic the code for `sqrt`. Have you tried that?

**Comment by krish8484** [comment]
Some unintentional changes are made due to version issues, I will close this one and fix it in my next PR.

**Comment by dpsanders** [comment]
There's no need to close and reopen. Just force push to this branch with

`git push -f`

File: Project.toml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 name = "IntervalArithmetic"
 uuid = "d1acc4aa-44c8-5952-acd4-ba5d80a2a253"
 repo = "https://github.com/JuliaIntervals/IntervalArithmetic.jl.git"
-version = "0.17.2"
+version = "0.17.0"
```
Comment (dpsanders) [inline Project.toml]: This isn't right. We should be increasing the version as you had before.

Comment (dpsanders) [reply Project.toml]: Each minor change like this should increment the "patch version", i.e. the 3rd number.

Comment (krish8484) [reply Project.toml]: In master, it is 0.17.1 so I initially changed it to 0.17.2 but the tests were conflicting so I reverted the commit.
Should I keep it at 0.17.2?

**Comment by krish8484** [comment]
@dpsanders

**Comment by krish8484** [comment]
@dpsanders

File: Project.toml (hunk 2 above)
Comment (dpsanders) [reply Project.toml]: Yes keep at 0.17.2.
What do you mean by "the tests were conflicting"?

**Comment by dpsanders** [comment]
You need to rebase on master I think and fix any conflicts.

From your branch do

`git rebase master`

It will tell you which files are in conflict. You have to fix those then add them as instructed, then do 
`git rebase --continue`

File: src/intervals/rounding.jl [hunk 3]
```diff
@@ -69,6 +69,10 @@ sqrt(a::T, rounding_mode::RoundingMode) where {T<:Rational} = setrounding(float(
     sqrt(float(a))
 end
 
+cbrt(a::T, rounding_mode::RoundingMode) where {T<:Rational} = setrounding(float(T), rounding_mode) do
```
Comment (dpsanders) [inline src/intervals/rounding.jl:72]: This `setrounding` stuff feels out of date. Is that what other functions do?

File: src/intervals/rounding.jl [hunk 4]
```diff
@@ -114,12 +118,15 @@ for T in (Float32, Float64)
 
         @eval sqrt(::IntervalRounding{:fast},
                             a::$T, $mode1) = sqrt_round(a, $mode2)
+
+        @eval cbrt(::IntervalRounding{:fast},
+                            a::$T, $mode1) = cbrt_round(a, $mode2)
```
Comment (dpsanders) [inline src/intervals/rounding.jl:123]: I don't think there is a `cbrt_round` function? Or is there?

File: test/interval_tests/numeric.jl [hunk 5]
```diff
@@ -353,6 +353,12 @@ end
         @test sqrt(big(2..3)) == Interval(big"1.414213562373095048801688724209698078569671875376948073176679737990732478462102", big"1.732050807568877293527446341505872366942805253810380628055806979451933016908815")
     end
 
+    @testset "cbrt" begin
+        @test cbrt(2..3) == Interval(1.259921049894873, 1.4422495703074085)
+
+        @test cbrt(big(2..3)) == Interval(big"1.259921049894873164767210607278228350570251464701507980081975112155299676513956", big"1.442249570307408382321638310780109588391869253499350577546416194541687596830003")
```
Comment (dpsanders) [inline test/interval_tests/numeric.jl:359]: Maybe you should check that the `big` version is contained inside the float version.
In general we should probably add tests like that to the tests.

**Comment by dpsanders** [comment]
Please don't use `git merge`.  Use `git rebase` as I mentioned.

**Comment by krish8484** [comment]
I have used `git rebase` and made the necessary changes.
Thanks @dpsanders

**Comment by dpsanders** [comment]
You can use `git rebase -i` ("interactive") to remove unnecessary commits and then force push.
There are various manuals online that explain the process. 

Alternatively you can just make a new branch with the correct commits, e.g. by `git cherry-pick`. 

None of this is really necessary for this PR, but they are useful skills to learn.

**Comment by dpsanders** [comment]
LGTM

**Comment by dpsanders** [comment]
Thanks!

**Comment by dpsanders** [comment]
Thanks!

**Pull request merged**